### PR TITLE
Membership page, site redesign

### PR DESCRIPTION
**Pull request by ClayMav**

### CODE REVIEW

File: ACM_General/core/templates/core/base.html [hunk 1]
```diff
@@ -1,85 +1,52 @@
-<!DOCTYPE html>
-{% load static %}
-
-{% block html_tag %}<html lang='en'>{% endblock %}
+<!DOCTYPE html>{% load static %}
+{% block html_tag %}
+  <html lang='en'>{% endblock %}
```
Comment (kevinschoonover) [inline ACM_General/core/templates/core/base.html:3]: Would there be any way to have the block start and end in a more stylish way. I understand from a html perspective 
this makes more sense with the formatting, but doesn't look good.

File: ACM_General/core/templates/core/base.html [hunk 2]
```diff
@@ -1,85 +1,52 @@
...
+      <meta name="author" content="ACM-MST SIG.com"/>{% endblock %}
+    {% block head_robots %}
+      <meta name="robots" content="NONE,NOARCHIVE"/>{% endblock %}
+    {% block head_base_stylesheets %}{% endblock %}
+    {% block head_stylesheets %}{% endblock %}
+    {% block head_javascript %}{% endblock %}
+    {% block head_extra-descriptors %}{% endblock %}
+    {% block head_fonts %}{% endblock %}
+    {% block head_title %}
+      <title> ACM | General </title>{% endblock %}{% endblock %}
```
Comment (kevinschoonover) [inline ACM_General/core/templates/core/base.html:41]: I see this ends head_defaultcontent but I still struggle with this formatting.

File: ACM_General/home/static/home/css/officers.css [hunk 3]
```diff
@@ -1,148 +1,135 @@
...
-    overflow: hidden;
-    position: relative;
-    transition: filter 0.5s cubic-bezier(.43,.41,.22,.91);
-    max-width: 155px;
-    min-width: 155px;
-    height: 100%;
+  background-position: center center;
+  background-repeat: no-repeat;
+  background-size: cover;
+  /*border-top-left-radius: 0.25rem;*/
```
Comment (kevinschoonover) [inline ACM_General/home/static/home/css/officers.css:82]: If these comments are necessary in the CSS, could you please put a little blurb about them explaining why they are commented out.

File: ACM_General/home/static/home/css/officers.css [hunk 4]
```diff
@@ -1,148 +1,135 @@
...
+/ / filter: saturate(180 %);
+  overflow: hidden;
+  position: relative;
+  transition: filter 0.5s cubic-bezier(.43, .41, .22, .91);
+  max-width: 155px;
+  min-width: 155px;
+  height: 100%;
 }
 
 .contain {
```
Comment (kevinschoonover) [inline ACM_General/home/static/home/css/officers.css:94]: These three blocks are completely commented out, semantic or other purpose? If important, please comment.

File: ACM_General/home/templates/home/base.html [hunk 5]
```diff
@@ -1,128 +1,127 @@
...
-
-{% block html_tag %}<html class="no-js" lang="en">{% endblock %}
-
+{% extends 'core/base.html' %}{% load static %}
+{% block html_tag %}
+  <html class="no-js" lang="en">{% endblock %}
 {% block head_title %}
-    <title> ACM | General </title>
-{% endblock %}
+  <title> ACM | General </title>{% endblock %}
```
Comment (kevinschoonover) [inline ACM_General/home/templates/home/base.html:5]: Change "ACM | General" to just "ACM" for default. Makes more semantic sense.

File: ACM_General/home/templates/home/base.html [hunk 6]
```diff
@@ -1,128 +1,127 @@
...
-        <!--(function(b,o,i,l,e,r){b.GoogleAnalyticsObject=l;b[l]||(b[l]=-->
-        <!--function(){(b[l].q=b[l].q||[]).push(arguments)});b[l].l=+new Date;-->
-        <!--e=o.createElement(i);r=o.getElementsByTagName(i)[0];-->
-        <!--e.src='https://www.google-analytics.com/analytics.js';-->
-        <!--r.parentNode.insertBefore(e,r)}(window,document,'script','ga'));-->
-        <!--ga('create','UA-XXXXX-X','auto');ga('send','pageview');-->
-    <!--</script>-->
-
-{% endblock %}
+  <script>
```
Comment (kevinschoonover) [inline ACM_General/home/templates/home/base.html:12]: Can you put a little description explaining this code and what it does.

File: ACM_General/home/templates/home/base.html [hunk 7]
```diff
@@ -1,128 +1,127 @@
...
+      <li class="nav-home nav-button"><a href="{% url "home:index" %}"
+                                         title="Home">
+        <img src="{% static "home/img/core/home-icon.png" %}"/>
+        <span>Home</span>
+      </a></li>
+      <li class="nav-sponsors nav-button"><a href="{% url "home:sponsors" %}"
+                                             title="Sponsors">
+        <img src="{% static "home/img/core/sponsors-icon.png" %}"/>
+        <span>Sponsors</span>
+      </a>
```
Comment (kevinschoonover) [inline ACM_General/home/templates/home/base.html:92]: No ending `</li>`

File: ACM_General/home/templates/home/base.html [hunk 8]
```diff
@@ -1,128 +1,127 @@
...
+            href="{% url "thirdparty_auth:login" "oauth2" "google" %}">Login</a>
+        </li>
+      </ul>
+    </div>
+  </div><!-- mobile menu -->
 
+  <div class="pseudo-nav"></div>
+  <nav><!-- desktop nav -->
+    <ul class="nav-link-wrapper">
+      <li class="nav-home nav-button"><a href="{% url "home:index" %}"
```
Comment (kevinschoonover) [inline ACM_General/home/templates/home/base.html:83]: Any way to format these as multiple lines? I don't think it looks good stylistically

File: ACM_General/home/templates/home/media.html [hunk 9]
```diff
@@ -3,68 +3,86 @@
...
+
+        <div class="m-p-g__fullscreen"></div>
+      </div>
     </div>
-</div>
+  </main>
 
-<script>
-	var elem = document.querySelector('.m-p-g');
+  <script>
```
Comment (kevinschoonover) [inline ACM_General/home/templates/home/media.html:50]: Another little blurb about what the function is doing.

File: ACM_General/home/templates/home/media.html [hunk 10]
```diff
@@ -3,68 +3,86 @@
...
-{#    <a href="{% static "/home/img/media/network_talk.jpg" %}" style="background-image: {% static "/home/img/media/network_talk.jpg" %}">#}
-{#        <img class="media-picture" src="{% static "/home/img/media/network_talk.jpg" %}" />#}
-{#    </a>#}
-{#    <a href="{% static "/home/img/media/sigcomp_1.jpg" %}" style="background-image: {% static "/home/img/media/sigcomp_1.jpg" %}">#}
-{#        <img class="media-picture" src="{% static "/home/img/media/sigcomp_1.jpg" %}" />#}
-{#    </a>#}
-{#    <a href="{% static "/home/img/media/icpc.jpg" %}" style="background-image: {% static "/home/img/media/icpc.jpg" %}">#}
-{#        <img class="media-picture" src="{% static "/home/img/media/icpc.jpg" %}" />#}
-{#    </a>#}
+  {#    <a href="{% static "/home/img/media/ctf.jpg" %}" style="background-image: {% static "/home/img/media/ctf.jpg" %}">#}
```
Comment (kevinschoonover) [inline ACM_General/home/templates/home/media.html:59]: Why are these all commented out?

File: ACM_General/home/views.py [hunk 11]
```diff
@@ -32,7 +32,14 @@ def media(request):
 
 
 def officers(request):
-    return(render(
+    return (render(
```
Comment (kevinschoonover) [inline ACM_General/home/views.py:35]: should really be

```
return(
    render(
       request,
       'home/officers.html'
    )
)

File: ACM_General/home/views.py [hunk 12]
```diff
@@ -32,7 +32,14 @@ def media(request):
...
         request,
         'home/officers.html',
     ))
+
+
+def membership(request):
```
Comment (kevinschoonover) [inline ACM_General/home/views.py:41]: See above.

**Comment by kevinschoonover** [comment]
Some more general things:
1. I'm really not a fan of the white color scheme being persistent in both the navbar and the footer. I personally don't thing it blends well nor transitions when moving from the header and footer to the body and looks all the same. Moreover, I am not a fan of the gray sponsor buttons on the white background nor the white background on the different nth child styles you have (like the event page and bottom of membership page). I don't mean to be super harsh, but we definitely can discuss if you enjoy this style.

2. I do really like the new buttons.

3. Can you change the buttons on the membership page to interact with Stripe and send a POST request to /payment-handler/ like in my example. I don't know if it is still committed, but if its not then let me know. (https://stackoverflow.com/questions/36659471/modify-style-of-pay-with-card-stripe-button)

4. I do like the new navbar, but just something to test out would be instead of only expanding the underlining when clicked, maybe have it transition to green as well.

File: ACM_General/core/templates/core/base.html (hunk 1 above)
Comment (ClayMav) [reply ACM_General/core/templates/core/base.html:3]: Looks like we have two options, first would be to use some **middleware** (namely [django-htmlmin](https://pypi.python.org/pypi/django-htmlmin)), and the second would be **spaceless tags**:
```
{% spaceless %}
{% endspaceless %}
```

File: ACM_General/core/templates/core/base.html (hunk 2 above)
Comment (ClayMav) [reply ACM_General/core/templates/core/base.html:41]: Honestly I don't like it either, from our end it is horrid, and I think it is less important to have it look good on the clients end if it looks bad to the developers, I will remove some of the de-whitespacing I did. It is very strange how Django adds up to 2 lines of whitespace per tag.

**Pull request merged**